include/deal.II/sundials/n_vector.templates.h

```diff
@@ -0,0 +1,806 @@
+{
+  auto *x_dealii = unwrap_nvector_const<VectorType>(x);
+  auto *z_dealii = unwrap_nvector<VectorType>(z);
+
+  AssertDimension(x_dealii->size(), z_dealii->size());
+
+  std::transform(x_dealii->begin(),
+                 x_dealii->end(),
+                 z_dealii->begin(),
+                 [](typename VectorType::value_type v) {
+                   return std::fabs(v);
+                 });
+}
+
+
+
+template <typename VectorType,
+          std::enable_if_t<IsBlockVector<VectorType>::value, int>>
+void
+SUNDIALS::internal::NVectorOperations::elementwise_abs(N_Vector x, N_Vector z)
+{
+  auto *x_dealii = unwrap_nvector_const<VectorType>(x);
+  auto *z_dealii = unwrap_nvector<VectorType>(z);
+
+  AssertDimension(x_dealii->size(), z_dealii->size());
+  AssertDimension(x_dealii->n_blocks(), z_dealii->n_blocks());
+
+  for (unsigned i = 0; i < x_dealii->n_blocks(); ++i)
+    std::transform(x_dealii->block(i).begin(),
+                   x_dealii->block(i).end(),
+                   z_dealii->block(i).begin(),
+                   [](typename VectorType::value_type v) {
+                     return std::fabs(v);
+                   });
+}
+
+
+
+template <typename VectorType>
+N_Vector
+SUNDIALS::internal::create_empty_nvector()
+{
+  N_Vector v = N_VNewEmpty();
+  if (v == nullptr)
+    return (nullptr);
+
+  /* constructors, destructors, and utility operations */
+  v->ops->nvgetvectorid = NVectorOperations::get_vector_id;
+  v->ops->nvclone       = NVectorOperations::clone<VectorType>;
+  v->ops->nvcloneempty  = NVectorOperations::clone_empty;
+  v->ops->nvdestroy     = NVectorOperations::destroy<VectorType>;
+  //  v->ops->nvspace           = undef;
+  v->ops->nvgetcommunicator = NVectorOperations::get_communicator<VectorType>;
+  v->ops->nvgetlength       = NVectorOperations::get_global_length<VectorType>;
+
+  /* standard vector operations */
+  v->ops->nvlinearsum = NVectorOperations::linear_sum<VectorType>;
+  v->ops->nvconst     = NVectorOperations::set_constant<VectorType>;
+  v->ops->nvprod      = NVectorOperations::elementwise_product<VectorType>;
+  v->ops->nvdiv       = NVectorOperations::elementwise_div<VectorType>;
+  v->ops->nvscale     = NVectorOperations::scale<VectorType>;
+  v->ops->nvabs       = NVectorOperations::elementwise_abs<VectorType>;
+  v->ops->nvinv       = NVectorOperations::elementwise_inv<VectorType>;
+  v->ops->nvaddconst  = NVectorOperations::add_constant<VectorType>;
+  v->ops->nvdotprod   = NVectorOperations::dot_product<VectorType>;
+  v->ops->nvmaxnorm   = NVectorOperations::max_norm<VectorType>;
+  v->ops->nvwrmsnorm  = NVectorOperations::weighted_rms_norm<VectorType>;
+  //  v->ops->nvwrmsnormmask = undef;
+  //  v->ops->nvmin          = undef;
+  //  v->ops->nvwl2norm      = undef;
+  //  v->ops->nvl1norm       = undef;
+  //  v->ops->nvcompare      = undef;
+  //  v->ops->nvinvtest      = undef;
+  //  v->ops->nvconstrmask   = undef;
+  //  v->ops->nvminquotient  = undef;
```

Comment:
For reference, I put some further operations that could be implemented here. In a quick test with the ARKode tests they didn't seem to be necessary.